src/store/mod.rs

```diff
@@ -49,6 +51,109 @@ pub trait StoreCapabilities {
     fn get_retention(&self) -> Option<&Self::Retention>;
 }
 
+/// Data passed to the store that identifies a property
+pub struct PropertyMapping<'a> {
+    interface_info: PropertyInterface<'a>,
+    path: &'a str,
+}
+
+impl<'a> PropertyMapping<'a> {
+    pub(crate) fn with_property_path(property_ref: &'a PropertyRef<'a>, path: &'a str) -> Self {
+        Self {
+            interface_info: property_ref.into(),
+            path,
+        }
+    }
+
+    pub(crate) fn new_unchecked(interface_info: PropertyInterface<'a>, path: &'a str) -> Self {
+        Self {
+            interface_info,
+            path,
+        }
+    }
```

Comment:
I would use a single method like the following:


```suggestion
    pub(crate) fn new_unchecked(interface_info: impl Into<PropertyInterface<'a>>, path: &'a str) -> Self {
        Self {
            interface_info: interface_info.into(),
            path,
        }
    }
```

Reply:
No, I'd prefer to keep them separated since `new_unchecked` could be considered an `unsafe` method because you have to make sure that the interface you are passing is actually a property interface but we have no way of knowing. At the same time `with_property_path` is checked by our other methods to receive an input property interface.
This is only used inside the crate so it's not a big deal, but it would be better to keep them separated IMO. If this doesn't make sense to most of us I can remove it.